bundles/async_remote_services/discovery/discovery_spi/CMakeLists.txt

```
@@ -0,0 +1,29 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+add_library(discovery_spi STATIC
+        src/do_not_remove.cc)
+
+set_target_properties(discovery_spi PROPERTIES OUTPUT_NAME "celix_discovery_spi")
+target_include_directories(discovery_spi PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+        $<INSTALL_INTERFACE:include/celix/discovery_spi>)
```

This can be a INTERFACE library
https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries

```suggestion
add_library(discovery_spi INTERFACE)
target_include_directories(discovery_spi INTERFACE
        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
        $<INSTALL_INTERFACE:include/celix/discovery_spi>)
```

Also IMO a single interface library rsa_spi is enough.
This could be named `rsa_spi`, but first rename the C rsa_spi (and refactor the CMakelists of the C RSA):
https://github.com/apache/celix/tree/master/bundles/remote_services/rsa_spi
to deprecated_rsa_spi.

---

The IEndpoint.h should also be included in the interface library.

---

Can the IEndpointListener interface also be added. I know it is not used atm, but I prefer the already have the inferface available. 

C++
```
namespace celix::rsa {
    class IEndpointListener {
    public:
         static constexpr std::string_view NAME{"celix::rsa::IEndpointListener"};
         static constexpr std::string_view VERSION{"1.0.0"};

          virtual ~IEndpointListener() noexcept = default;
           addDiscoveredEndpoint(const std::shared_ptr<const celix::rsa::IEndpoint>& endpoint);
           removeDiscoveredEndpoint(const std::shared_ptr<const celix::rsa::IEndpoint>& endpoint);
    };
}
```

---

What is the use of an `IEndpointListener` if we already have trackers?